src/Common/src/TypeSystem/IL/ILImporter.cs

```diff
@@ -905,12 +928,18 @@ private void ImportBasicBlock(BasicBlock basicBlock)
                 EndImportingInstruction();
 
                 // Check if control falls through the end of method.
-                if (_currentOffset >= _basicBlocks.Length)
+                if (_currentOffset == _basicBlocks.Length)
                 {
                     ReportFallthroughAtEndOfMethod();
                     return;
                 }
 
+                if (_currentOffset > _ilBytes.Length)
```

Comment:
This check should not be needed anymore - `_currentOffset > _ilBytes.Length` should be always false here because of we check for it upfront now.